### PR TITLE
D 16204

### PR DESCRIPTION
**Pull request by walkerrandolphsmith**
Multiselect has several bugs similar to the ones recently fixed in Single Select.

User input (typing) should be enabled. Filtering of the options is configurable but defaults to string matching on the "label"

The component now takes a collection of item objects as props. As long as they have a `value` and `label` then the items can be filtered and kept track of.

Remove prop spreading

### CODE REVIEW

File: packages/components/src/MultiSelect/MultiSelect.js [hunk 1]
```diff
@@ -1,74 +1,143 @@
...
     action(selectedId, event);
 
-    this.props.clearInput();
+    return {
+      isOpen: false,
+      inputValue: '',
+    };
   }
 
+  handleChanger = (...rest) => {
```
Comment (klmarsh4) [inline packages/components/src/MultiSelect/MultiSelect.js]: This entire handler does not seem like something we need. is it?

File: packages/components/src/MultiSelect/MultiSelect.js [hunk 2]
```diff
@@ -79,287 +148,144 @@ class MultiDownshift extends React.Component {
...
+            <Downshift
+              onChange={this.handleChanger}
+              disabled={disabled}
+              stateReducer={this.stateReducer}
+              selectedItem={selectedItems}
+              itemToString={item => (item ? item.label : '')}
+              initialInputValue=""
+            >
+              {downshift => {
+                console.log('inputValue', downshift.inputValue);
```
Comment (klmarsh4) [inline packages/components/src/MultiSelect/MultiSelect.js]: is this console log still needed?

File: packages/components/src/MultiSelect/MultiSelect.js [hunk 3]
```diff
@@ -79,287 +148,144 @@ class MultiDownshift extends React.Component {
...
+                  onBlur: handleBlur,
+                  onFocus: handleFocus,
+                  calculateDropdownDimensions,
+                  focused: calculatedFocused,
+                  onMouseEnter,
+                  onMouseLeave,
+                  hovered: calculatedHovered,
+                  inputContainerRef,
+                  inputRef,
+                  getItems: this.getItems,
```
Comment (klmarsh4) [inline packages/components/src/MultiSelect/MultiSelect.js]: with this many props it might be helpful if we alphabetized them

File: packages/components/src/MultiSelect/renderField.js [hunk 4]
```diff
@@ -0,0 +1,431 @@
...
+    removeItem,
+    handleSelection,
+    isOpen,
+    getItemProps,
+    highlightedIndex,
+    openMenu,
+    closeMenu,
+    inputValue,
+
+    getItems,
```
Comment (klmarsh4) [inline packages/components/src/MultiSelect/renderField.js]: again, might be good to have some order to these props

File: packages/components/src/MultiSelect/renderField.js [hunk 5]
```diff
@@ -0,0 +1,431 @@
...
+  } = props;
+
+  console.log('render field', selectedItems, inputValue, isOpen);
```
Comment (klmarsh4) [inline packages/components/src/MultiSelect/renderField.js]: is this needed

File: packages/components/src/Typography/Typography.js [hunk 6]
```diff
@@ -57,6 +57,7 @@ class Typography extends Component {
         ),
       }),
       is,
+      ['children'],
```
Comment (klmarsh4) [inline packages/components/src/Typography/Typography.js]: These changes seem unrelated. Are they needed?

**Comment by CWSites** [comment]
Just an FYI, I was reviewing with @klmarsh4. Also this and #123 appear to be identical, can we close one of them?

File: packages/components/src/MultiSelect/MultiSelect.js (hunk 1 above)
Comment (walkerrandolphsmith) [reply packages/components/src/MultiSelect/MultiSelect.js]: Oh no it was for debugging. Sloppy. Thanks

File: packages/components/src/MultiSelect/MultiSelect.js (hunk 2 above)
Comment (walkerrandolphsmith) [reply packages/components/src/MultiSelect/MultiSelect.js]: More sloppiness.
